atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -141,26 +195,24 @@ def run(self):
         except KeyError:
             release = None
 
+        # Always set preferred release label - other will be set if old-style
+        # label is present
+        release_label = labels.LABEL_NAMES[Labels.LABEL_TYPE_RELEASE][0]
+
         if release:
             if not self.append:
                 self.log.debug("release set explicitly so not incrementing")
                 if not is_scratch_build():
                     self.check_build_existence_for_explicit_release(component, version, release)
-                return
+                    dockerfile_labels[release_label] = release
+                else:
+                    return
 
-        if self.append:
-            next_release = self.get_next_release_append(component, version, release)
-        elif is_scratch_build():
-            metadata = get_build_json().get("metadata", {})
-            next_release = metadata.get("name", "1")
-        else:
-            next_release = self.get_next_release_standard(component, version)
+        if not release or self.append:
+            self.next_release_general(component, version, release, release_label,
+                                      dockerfile_labels)
 
-        # Always set preferred release label - other will be set if old-style
-        # label is present
-        release_label = labels.LABEL_NAMES[Labels.LABEL_TYPE_RELEASE][0]
+        if not self.predeclare_build or is_scratch_build():
```

Comment:
you can invert this statement and get the final line inside the block (no need for `return`)

`if self.predeclare_build and not is_scratchbuild():`

Reply:
sure, now when we call just 1 function in there makes sense